lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -118,12 +118,9 @@ e.g. 'tag:0123:important:GM' returns a list of '0123', 'important' and 'GM'.
 =cut
 sub tag {
     my ($self) = @_;
-    $self->text =~ /\btag:(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[@\d\w]+))?\b/;
-    # note: 'build' here may be either VERSION-BUILD or just BUILD.
-    # If the tag is in VERSION-BUILD form it will return the entire
-    # value, it does not isolate the BUILD component. Things that
-    # consume it must be aware of this
-    return $+{build}, $+{type}, $+{description};
+    $self->text
+      =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[@\d\w]+))?\b/;
```

Comment:
please crosscheck with the last change in this line in https://github.com/Martchus/openQA/commit/4d712dfd80e30d1f0850c6b76c387509e39d0f6f , just to be sure :-)